x-pack/plugins/security_solution/public/transforms/utils/get_transform_changes_if_they_exist.ts

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { adjustTimeRange } from './adjust_timerange';
+import { getSettingsMatch } from './get_settings_match';
+import { getTransformChanges } from './get_transform_changes';
+import { isFilterQueryCompatible } from './is_filter_query_compatible';
+import { GetTransformChangesIfTheyExist } from './types';
+
+// TODO: Add the other switches here such as the disabling of a widget/factory type
+// or if a transform is disabled, then this cannot use the query
+export const getTransformChangesIfTheyExist: GetTransformChangesIfTheyExist = ({
+  factoryQueryType,
+  indices,
+  transformSettings,
+  filterQuery,
+  histogramType,
+  timerange,
+}) => {
+  if (!transformSettings.enabled) {
+    // Early return if we are not enabled
+    return { factoryQueryType, indices, timerange };
+  }
```

Comment:
Nice abstraction! Will make UI/UX implementation much cleaner with having each client query go through this check 👍